build_tag_push.py

```diff
@@ -1,101 +1,130 @@
+	image_name = parseAppFile(filename)['name'] # Name from app.yml file
+	subprocess.check_call(["docker", "build", docker_dir_context, "-t", image_name]) # this builds the docker image.
+
+	# QUESTION: When I try to build the image from the file that has the docker context from the app.yml file 
+	# then I get the error: "Build an image from a Dockerfile". The only way I was able to complete this was 
+	# putting in my own Dockerfile, which I am assuming the groups will have as well and we will need to copy
+	# this over into that directory created, or is this going to be copied over manually?
+
+# Step 4 - Tags the image with the current commit hash & push to Docker Hub = COMPLETE.
+
+def tag_DockerImage(filename):
+	'''
+	This function is finding the tagging the Docker Image with the name of the user/image_name, and the current git hash. 
+	Note: The current git hash is shortened a bit, but to display the full hash, then change the lower case %h to upper case %H. 
+	The source for this command is: https://container-solutions.com/tagging-docker-images-the-right-way
+	'''
+	image_name = parseAppFile(filename)['name']
+
+	subprocess.check_call(["docker", "tag", image_name, Docker_Hub_Name+'/'+image_name]) # Need to tag the image with the namespace for the Docker Hub account. 
+																						 # Note: I made this a global variable for now with a comment for user input if needed.
+
+	current_hash = str(subprocess.check_call(["git", "log", "-1", "--pretty=%h"])) # Note: having the lower case 'h' provides a shorter version of the hash... 
+	 																			   # I read that the shorter tag was better so I did that here.
+
+	subprocess.check_call(["docker", "tag", image_name, Docker_Hub_Name+'/'+image_name+':'+current_hash]) # This tags the image with the current hash.
+
+	# QUESTION: When I am running this function, the current_hash is turning to the number 0, and not what the actual current_hash is when observed in the terminal (which is 07f21e2), 
+	# I have been having this problem a lot with setting a variable = terminal command... it always becomes 0. Is there something I am doing wrong or missing with this?
+
+def push_DockerImage(filename):
+	'''
+	This function is to push the docker image once it has been built and tagged with the name, and the commit hash tag, which is dependent on the 
+	functions build_DockerImage and tag_DockerImage functions.
+	'''
+	image_name = parseAppFile(filename)['name']
+	subprocess.check_call(["docker", "push", Docker_Hub_Name+'/'+image_name])
 
 if __name__ == '__main__':
 	#filename = "test.yaml"
 	filename = "ymltest.yml"
-	data = parseAddFile(filename)
-	# print data # This will print the returned yaml file information in the form of a list
-	build_DockerImageandTag(filename)
\ No newline at end of file
+	data_from_appyaml = parseAppFile(filename)
+	createContextDirectory(parseAppFile(filename)['docker_context'])
```

Comment:
here instead of `parseAppFile(filename)['docker_context']` you could `data_from_appyaml['docker_context']`. And when you call the methods below you could pass `data_from_appyaml` and you wouldn't have to keep re-parsing the yaml file.